.github/workflows/smoke-test.yml

```diff
@@ -0,0 +1,81 @@
+# Boot Smoke Test only tests Bare VM (start,stop,delete) and not any minikube/kubernetes functionality.
+name: Boot Smoke Test
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+  workflow_dispatch:
+env:
+  GOPROXY: https://proxy.golang.org
+  GO_VERSION: '1.24.0'
+  permissions:
+  contents: read
+
+jobs:
+  minikube-vm-boot:
+    name: Boot Smoke Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-13]
+        driver: [qemu, vfkit]
+    steps:
+      - name: Info Block
+        shell: bash
+        run: |
+          uname -a
+          sysctl -n hw.memsize
+          echo "$(sysctl -n hw.memsize) / 1024 / 1024 / 1024" | bc
+          sysctl -n hw.ncpu
+          sysctl -n machdep.cpu.brand_string
+          sysctl hw.model
+          sysctl -n kern.hv_vmm_present
+          sysctl -n kern.hv_support
+          system_profiler SPHardwareDataType
+          sw_vers
+          ifconfig
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+      - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5
+        with:
+          go-version: ${{env.GO_VERSION}}
+      - name: Download Dependencies
+        run: go mod download
+      - name: Build Binaries
+        run: make
+      - name: Ensure bootpd is enabled
+        shell: bash
+        run: |
+          set -x
+          fw=/usr/libexec/ApplicationFirewall/socketfilterfw
+          sudo $fw --remove /usr/libexec/bootpd
+          sudo $fw --add /usr/libexec/bootpd
+          sudo $fw --unblock /usr/libexec/bootpd
+      - name: Install vfkit
+        if: matrix.driver == 'vfkit'
+        run: |
+          brew update
+          brew install vfkit
+      - name: Install qemu and socket_vmnet 
+        if: matrix.driver == 'qemu'
+        run: |
+          brew update
+          brew install qemu
+          brew install socket_vmnet
+          HOMEBREW=$(which brew) && sudo ${HOMEBREW} services start socket_vmnet          
+      - name: Start minikube (1st boot)
+        run: |
+          set -ex
```

Comment:
-e is the default, and this is one command so we don't really need -x. I would be nicer as:

    - name: ...
      run: out/minikube start ...